File: solutions/studyplan/programmingskills/easy_66/Solution.java
package studyplan.programmingskills.easy_66;

import java.lang.reflect.Array;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Objects;

public class Solution {

    /*  My Solution
     *  Runtime: 0ms
     *  Memory: 43.30MB
     */
    public static int[] plusOne(int[] digits) {

        for (int i = digits.length - 1; i >= 0; i--) {
            if (digits[i] < 9) {
                digits[i]++;
                return digits;
            }
            digits[i] = 0;
        }
        int[] result = new int[digits.length + 1];
        result[0] = 1;
        return result;
    }


}

File: solutions/studyplan/programmingskills/easy_66/Solution.kt
package studyplan.programmingskills.easy_66

import java.sql.Array

/*  My Solution
   *  Runtime: 10ms
   *  Memory: 44.06MB
   */
fun plusOne(digits: IntArray): IntArray {
    for (i in digits.lastIndex downTo 0) {
        if (digits[i] < 9) {
            digits[i] += 1
            return digits
        }
        digits[i] = 0
    }
    val result = IntArray(digits.size + 1)
    result[0] = 0
    return digits
}

fun main() {
    println(plusOne(intArrayOf(1, 2, 3, 4)))
}